snapcraft/commands/names.py

```diff
@@ -138,12 +149,25 @@ def run(self, parsed_args):
         if not snaps:
             emit.message("No registered snaps")
         else:
-            tabulated_snaps = tabulate(
-                snaps,
-                headers=["Name", "Since", "Visibility", "Notes"],
-                tablefmt="plain",
-            )
-            emit.message(tabulated_snaps)
+            headers = ["Name", "Since", "Visibility", "Notes"]
+
+            if parsed_args.format == "table":
+                tabulated_snaps = tabulate(
+                    snaps,
+                    headers=headers,
+                    tablefmt="plain",
+                )
+                emit.message(tabulated_snaps)
+            elif parsed_args.format == "json":
+                json_snaps = {
+                    "snaps": [
+                        {header.lower(): snap[n] for n, header in enumerate(headers)}
+                        for snap in snaps
+                    ]
+                }
+                emit.message(json.dumps(json_snaps))
+            else:
+                raise NotImplementedError("Format not implemented", parsed_args.format)
```

Comment:
Can you add a unit test for this too?

Something along the lines of
```python
 def test_table_error(self, emitter, fake_store_get_names, command_class):
        cmd = command_class(None)

        with pytest.raises(NotImplementedError) as raised:
            cmd.run(
                argparse.Namespace(
                    store_id="1234",
                    private=False,
                    yes=True,
                    format="invalid",
                    **{"snap-name": "test-snap"},
                )
            )

         assert raised == NotImplementedError("Format not implemented", "invalid")
```

Reply:
@mr-cal Done :)  I also updated the original functionality to pretty print the json using `indent=4`. `pytest` & `tox` both pass successfully.

The tests have kicked off again which is a good sign.